internal/azure/azure.go

```diff
@@ -163,6 +163,50 @@ func (c *Client) GetStorageAccessKeys(ctx context.Context, resourceGroup, storag
 	}, nil
 }
 
+func (c *Client) ListResourceGroups(ctx context.Context) ([]*armresources.ResourceGroup, error) {
+	client, err := c.newResourceClientFactory()
+	if err != nil {
+		return nil, err
+	}
+
+	pager := client.NewResourceGroupsClient().NewListPager(nil)
+
+	var groups []*armresources.ResourceGroup
+
+	for pager.More() {
+		page, err := pager.NextPage(ctx)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list resource groups: %w", err)
+		}
+
+		groups = append(groups, page.Value...)
+	}
+
+	return groups, nil
+}
+
+func (c *Client) TestHostedZoneLivenessWildcard(ctx context.Context, domainName string) (bool, *string, error) {
+	groups, err := c.ListResourceGroups(ctx)
+	if err != nil {
+		return false, nil, err
+	}
+
+	// Search through resource groups and return true for first match
+	for _, resourceGroup := range groups {
+		name := resourceGroup.Name
+		hasDomain, err := c.TestHostedZoneLiveness(ctx, domainName, *name)
+		if err != nil {
```

Comment:
we need to test for all domains and should continue if we didn't find for first one ,this should be removed ?

Reply:
I don't quite understand what you mean here - please explain further

Reply:
as we are searching for first true ,we should not return false if we there was an error for some group, we should return false once we have checked for all groups

Reply:
Unless I change the type to `*bool` (which I don't want to do), I can only return `true` or `false` - the previous [iteration of this](https://github.com/konstructio/kubefirst-api/blob/main/internal/azure/azure.go#L175) returned `bool`